packages/react-native/ReactCommon/react/renderer/core/RawValue.h

```diff
@@ -96,73 +111,157 @@ class RawValue {
    */
   template <typename T>
   explicit operator T() const {
-    return castValue(dynamic_, (T*)nullptr);
+    if (std::holds_alternative<folly::dynamic>(value_)) {
+      folly::dynamic dynamic = std::get<folly::dynamic>(value_);
+      return castValue(dynamic, (T*)nullptr);
+    } else {
+      const auto& [runtime, value] = std::get<JsiValuePair>(value_);
+      return castValue(runtime, value, (T*)nullptr);
+    }
   }
 
   inline explicit operator folly::dynamic() const noexcept {
-    return dynamic_;
+    return std::get<folly::dynamic>(value_);
   }
```

Comment:
Also, this function can no longer be `noexcept` since `std::get` is not `noexcept`.

Reply:
Ah yes, that's even worse. That means we won't get an exception, but the application will immediately abort.

Reply:
Hm my best idea for this would be to add a `react_native_assert` here so that during development there will be a receivable error?

Once this PR landed I will open a second PR that shows how to use the Jsi props without turning on the feature flag on a per component basis that would probably help with migration (and this specific call side).

Reply:
hm, or we could actually do a convert call (jsi -> folly) in case we stored a jsi value

Reply:
Let's do `react_native_assert` so it fails in development, but doesn't crash in release builds.

Reply:
Just to be sure:

- Right now I added to convert the jsi value to folly::dynamic if this method is called while we store value as jsi value pair.
- You say it would be better to not convert but to throw in this case? (In this case turning on the feature flag would break the call-site; but that might not be a problem with the follow up PR I intent to open?)